Feat: post 페이지 기본 구현

ckswnskfk opened the pull request:
카드 컴포넌트 스케치? 포함

## 어떤 변경인지 
- [x] Feat: 기능 변경, 기능 추가
- [ ] Docs: 문서 작업
- [ ] Refactor: 기능 변경 없이 코드 수정
- [ ] Style: 디자인
- [ ] Fix: 버그 수정
- [ ] Test: 테스트 코드 작성

## 설명
포스트 페이지 레이아웃을 만들었습니다.

색상이나 이미지는 추후 데이터 fetching을 통해 변경하는 걸로 바꾸겠습니다.
페이지 레이아웃을 구현하기 위해선 카드 컴포넌트 또한 필요해서 카드 컴포넌트의 스케치? 또한 포함해 pr 합니다.
페이지 내부 컨텐츠가 카드 리스트 뿐이라 따로 컴포넌트로 뺄 이유가 없는 것 같아 페이지에 직접 작성했습니다만 의견 주세요!

_______________

양이 많아서 죄송합니다.. 머지가 밀리니 이런 일이 생기는군요! 😢 
dev에 머지된 내용 모두 rebase 해서 작업했습니다.
용준님께서 작업하신 Card 컴포넌트와 fetcher를 사용해 만들었습니다.
무한 스크롤 등은 추후 개발하겠습니다. 미디어 쿼리도 다시 확인해야 할 것 같구요.

부디 conflict가 나지 않기를 바랍니다..


### 이슈 번호
https://github.com/Codeit-Rolling-11-Letsgo/Rolling/issues/14

### To Reviewers

ckswnskfk committed:
Feat: post 페이지 레이아웃 구현

카드 컴포넌트 스케치? 포함

mizign commented:
미디어 쿼리정도는 공통 mixin으로 지정해두고 써도 편할 것 같다는 생각이 들어요! 굉장히 많은 scss 파일에서 사용하게 될 것 같아서 스타일에서 일관성도 챙기고 코드 양도 줄일 수 있을 것 같습니다. 일단 저희 1차 목표가 pc 버전을 완성시키는 거라 찬주님도 mobile 대응을 크게 신경쓰진 않으셨겠지만 제가 개인적으로 궁금했던 부분은 두가지 정도인데용,
1. mobile 페이지에서 카드가 시안과 달리 2행으로 나눠보여요
2. 시안에서 카드 크기가 각각 모바일, 태블릿, pc에서 각각 320, 352, 384px로 'fixed'이던데, 카드를 이렇게 말랑말랑(?)하게 하는게 기획 의도상 맞는건지 급 궁금했습니다. 저는 시안을 보고 태블릿 사이즈에서 카드가 352로 고정이고, 여백이 늘어나는 거라고 이해했었거든용.

mizign commented on src/pages/post/PostPage.jsx:
```diff
@@ -1,3 +1,20 @@
 export default function PostPage() {
-	return <div>포스트 페이지</div>;
+	return (
+		<PostLayout>
+			<div className={styles.cardList}>
```
이모지 바는 요 사이에 넣어서 나중에 합치게 되는 거 맞을까요?

ckswnskfk replied:
이모지 바는 용준님이 만드신 Layout 컴포넌트를 호출할 때 addon props로 넘겨주면 될 것 같아요!
```jsx
function PostLayout({ children }) {
	return (
		<Layout addon={이모지 바}>
			<div className={styles.container}>{children}</div>
		</Layout>
	);
}
```
정도가 될 것 같습니다

mizign replied:
그런데 addon은 로고가 포함된 gnb상에서 오른쪽에 위치한 컴포넌트라고 이해해서 
예) 아래 이미지에서 '롤링페이퍼 만들기' 버튼 컴포넌트 
![image](https://github.com/Codeit-Rolling-11-Letsgo/Rolling/assets/148179726/a39d29f1-b7fb-45ee-90ad-242bba3d8a7b)

제가 만들어야하는 이모지바는 본문에 포함을 시켜야하는게 아닌가 싶더라구요.

ckswnskfk replied:
아 그러네요?ㅋㅋㅋㅋ 😅 
이 자리도 하나 만들어야겠네요

mizign commented on src/pages/post/PostPage.module.scss:
```diff
@@ -0,0 +1,29 @@
+	}
+}
+
+@media only screen and (max-width: 768px) and (min-width: 361px) {
+	.cardList {
+		grid-template-columns: repeat(2, minmax(20rem, 1fr));
+		gap: 1.6rem;
+	}
+}
```
저는 항상 max width로 통일해서 작성했었는데, 다른 분들 스타일은 어떤지 궁금하네요👀

ckswnskfk commented:
@sozign 
> mobile 페이지에서 카드가 시안과 달리 2행으로 나눠보여요

아마 더 줄이시면 1열로 나올텐데 제가 break point를 잘못 이해한 것 같습니다ㅋㅋㅋ

>시안에서 카드 크기가 각각 모바일, 태블릿, pc에서 각각 320, 352, 384px로 'fixed'이던데, 카드를 이렇게 말랑말랑(?)하게 하는게 기획 의도상 맞는건지 급 궁금했습니다. 저는 시안을 보고 태블릿 사이즈에서 카드가 352로 고정이고, 여백이 늘어나는 거라고 이해했었거든용.

제가 요구사항에서 `PC에서 카드 목록 영역의 너비는 1200px로 고정하고 좌우 여백만 변하다가, 화면의 너비가 1248px보다 줄어들 때 좌우 여백은 24px로 고정하고 카드들의 너비가 줄어듭니다.` 이런 걸 봐서 모든 화면에서도 그럴 거라고 생각했나봐요. 뭐가 맞는지는 잘.. 😅

mizign commented:
> 제가 요구사항에서 `PC에서 카드 목록 영역의 너비는 1200px로 고정하고 좌우 여백만 변하다가, 화면의 너비가 1248px보다 줄어들 때 좌우 여백은 24px로 고정하고 카드들의 너비가 줄어듭니다.` 이런 걸 봐서 모든 화면에서도 그럴 거라고 생각했나봐요. 뭐가 맞는지는 잘.. 😅

아 제가 이 요구사항을 놓쳤네요.. PC에 이렇게 명시가 되어있었으면,, 찬주님 구현하신 방안이 맞을 것 같네요

ckswnskfk committed:
Feat: PostLayout에 이모지 바(pageTitle) prop 추가

ckswnskfk committed:
Style: PostLayout의 container height 속성 삭제

height 100% 설정으로 카드 수가 늘어나도 높이가 늘어나지 않는 문제 해결

mizign committed:
feat: DropDown 컴포넌트 제작 (#15)

* feat: DropDown 컴포넌트 제작

* feat: disabled, error 경우 대응

* fix: hover시 글자색 변경 적용

* fix: DropDown 폴더 생성

* docs: js doc 작성

* fix: 리뷰반영 (커밋 메세지 본문참조)

1. 파이어폭스, 사파리에서 일부 스타일이 적용되지 않는 오류 해결
2. onChange prop 추가
3. 일반/focus/active/hover 스타일 조정
4. focus시 border 두꺼워지며 컴포넌트 밀리는 현상 해결

* fix: 'disabled' prop default value 수정, 스타일 수정

* fix: 옵션 선택시 드롭다운 목록 닫히게

* fix: onChange 이벤트 발생하지 않던 오류 해결, state명 수정

* docs: dropDown js doc 수정

* refactor: clsx 사용, className 정돈

* refactor: disabled 상태 가상 클래스로 스타일링

* fix: 길이가 긴 옵션이 선택되었을 경우 옵션 텍스트가 컴포넌트 밖으로 나오는 현상 해결

* refactor: 리뷰사항 반영

* refactor: onChange 함수를 handleOptionSelect에 위치시킴

---------

Co-authored-by: So-Yeon Ahn <sozygn@gmail.com>

dkile committed:
Feat: 홈페이지 메인 컨텐츠 구현 (#20)

* Feat: 이미지 컴포넌트 추가

* Feat: 카드 컴포넌트 추가

* Feat: 홈페이지 구현

* Fix: 메인 컨텐츠 스크롤 생기는 이슈 수정

dkile committed:
Feat: 공통 api fetcher 추가 (#29)

mizign committed:
feat: EmojiPickerButton 컴포넌트 제작 (#27)

* feat: EmojiPickerButton 컴포넌트 제작

* review: 이모지 클릭시 토글 닫히는 기능 추가

---------

Co-authored-by: So-Yeon Ahn <sozygn@gmail.com>

ckswnskfk committed:
Feat: post 페이지 기본 구현

ckswnskfk commented on src/apis/post/postAPI.js:
```diff
@@ -0,0 +1,14 @@
+
+export const getMessages = async ({ recipientId, limit, offset }) => {
+	try {
+		const res = await fetcher.get(
+			`recipients/${recipientId}/messages/?limit=${limit}&offset=${offset}`,
+		);
```
일단 급한대로 이렇게 작성했는데 아무래도 더러워 보입니다.. 실제로 요청이 가는 것도 그렇구요

ckswnskfk commented on src/components/post/PostCard.jsx:
여기도 더 깔끔하게 작성할 수 있을 것 같은데 잘 모르겠습니다..
리뷰로 후드러 패주세요

ckswnskfk commented on src/components/post/PostCard.module.scss:
scss가 익숙치 않아 어떻게 잘 쓰는지 잘 모르겠네요. 한 수 알려주세요!

ckswnskfk commented on src/pages/post/PostPage.jsx:
무한 스크롤은 추후 반영하겠습니다.

ckswnskfk reviewed (commented):
rebase 한 파일들도 바뀐 파일로 나오는데, 이게 정상인지 잘 모르겠습니다.. 😅 
제가 작성한 파일중 리뷰가 필요한 곳에 코멘트 남겨뒀으니 리뷰 부탁드립니다.

ckswnskfk committed:
Feat: 날짜 포맷 변경 함수 추가
